### ISSUE TITLE
ENH Haar, Cauchy, ... wavelets

### ISSUE THREAD

**Issue by edouardoyallon** [kymatio/kymatio#321]
It would be great to propose several sets of wavelets. Gabor are real in fourier, thus adding Haar wavelets (despite being simple and having a different structure) could make a lot of sens w.r.t. the current implementation. Cauchy wavelets are also a natural candidate.

**Comment by rleonarduzzi** [comment]
I agree, and I can do this. I have an almost read patch with battle-lemarie and bump-steerable wavelets for 1d that hopefully I'll commit soon. I can add Haar, Cauchy and others.

**Comment by edouardoyallon** [comment]
Awesome! This will be a great feature.

**Comment by lostanlen** [comment]
I would prefer to make it in two stages: first, an API change, and then a feature enhancement 
otherwise this issue is never going to be closed, given how many possibilities there are
one important element in the discussion is that some wavelets are better suited for FFT-based convolutions whereas others (like Haar) are better suited to spatial convolutions
wouldn't it be a little bit strange (and slow) to have to compute an FFT and IFFT to compute a mere sum+difference in Haar?

**Comment by edouardoyallon** [comment]
that's ok, spatial convolutions will be a future feature but I thinkdoing everything in fourier is more than fine

**Comment by lostanlen** [comment]
As you prefer. It would be good to delimit the scope of this issue then: what belongs to v0.2, v0.3, and v1.0
This is a feature request so it cannot go into Kymatio v0.1

**Comment by janden** [comment]
The thing to keep in mind here is whether these other wavelets would be useful for some specific purpose. Remember that every line we add has to be maintained. I could see Haar and spline wavelets being very useful once we get spatial convolutions, for example, but am less sure about Cauchy.

**Comment by lostanlen** [comment]
Yes. This message in #322 describes how I did it in Julia. It ended up being very little code per wavelet, and was highly extensible. But I had to implement boolean traits regarding dyadicity and steerability of wavelets, as well as whether they were available in spatial domain, or Fourier domain, or both.

https://github.com/kymatio/kymatio/issues/322#issuecomment-457260599

**Comment by edouardoyallon** [comment]
@rleonarduzzi where are we?

**Comment by rleonarduzzi** [comment]
I'll try and make a pull request by the end of the week

**Comment by edouardoyallon** [comment]
hi @rleonarduzzi

**Comment by TimD1** [comment]
Any updates on the current status of the Haar wavelets? I would like to use it, but can't find in the docs if it's been added yet. Thanks!

**Comment by rleonarduzzi** [comment]
Sorry for dropping the ball on this for such a long time. I've been swamped on the new job. I'll get back to it asap.
